akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -1378,15 +1407,39 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
         // TODO split it to several DeltaPropagation if too many entries
         replica(node) ! deltaPropagation
     }
-    if (deltaPropagationSelector.propagationCount % deltaPropagationSelector.divisor == 0)
+
+    if (deltaPropagationSelector.propagationCount % deltaPropagationSelector.gossipIntervalDivisor == 0)
       deltaPropagationSelector.cleanupDeltaEntries()
   }
 
-  def receiveDeltaPropagation(deltas: Map[String, DataEnvelope]): Unit = {
-    if (log.isDebugEnabled)
-      log.debug("Received DeltaPropagation from [{}], containing [{}]", sender().path.address, deltas.keys.mkString(", "))
+  def receiveDeltaPropagation(fromNode: UniqueAddress, deltas: Map[String, Delta]): Unit = {
+    val isDebugEnabled = log.isDebugEnabled
+    if (isDebugEnabled)
+      // FIXME log.debug("Received DeltaPropagation from [{}], containing [{}]", fromNode.address,
+      //        deltas.collect { case (key, Delta(_, fromSeqNr, toSeqNr)) => s"$key $fromSeqNr-$toSeqNr" }.mkString(", "))
+      log.debug("Received DeltaPropagation from [{}], containing [{}]", fromNode.address,
+        deltas.collect { case (key, Delta(d, fromSeqNr, toSeqNr)) ⇒ s"$key $fromSeqNr-$toSeqNr ${d.data}" }.mkString(", "))
     deltas.foreach {
-      case (key, envelope) ⇒ writeAndStore(key, envelope)
+      case (key, Delta(envelope @ DataEnvelope(_: RequiresCausalDeliveryOfDeltas, _, _), fromSeqNr, toSeqNr)) ⇒
```

Comment:
Yes. Is the `deltas` `fromSeqNr` increased when we receive full update containing the already collected deltas?